internal/provider/resource_apply_module.go

```diff
@@ -554,6 +685,60 @@ func (t *applyModuleResource) getCurrentApplied(ctx context.Context,
 	return nil
 }
 
+// extractRunError extracts the error from a run's logs (if the run errored out).
+func (t *applyModuleResource) extractRunError(ctx context.Context, run *sdktypes.Run) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if run.Status == sdktypes.RunErrored {
+		var jobID string
+
+		switch {
+		case run.Apply != nil:
+			if run.Apply.CurrentJobID != nil {
+				jobID = *run.Apply.CurrentJobID
+			}
+		case run.Plan != nil:
+			if run.Plan.CurrentJobID != nil {
+				jobID = *run.Plan.CurrentJobID
+			}
+		}
+
+		if jobID == "" {
+			diags.AddWarning("Run status is errored, but no job ID found", "")
+			return diags
+		}
+
+		logs, err := t.client.Job.GetJobLogs(ctx, &sdktypes.GetJobLogsInput{ // unlimited
```

Comment:
This must iterate over the logs in chunks since unlimited is not supported. It can start at the end of the logs and paginate in reverse.

Reply:
I see two difficulties with starting at the end of the logs and paginating in reverse:
1. It appears that will require (at least) another SDK call to get the log size in order to know the offset of the end of the logs;
2. If we anticipate that there could be considerable distance from the first error to the end of the logs, we'll have to get all the logs back to the beginning anyway.
I would suggest starting at the beginning (offset=0) and paginating forward.